### PR TITLE
Update adversarial training code

### PR DESCRIPTION
**Pull request by asim29**
List of changes made: 
- Removed the division by 255 for epsilon. Directly setting an epsilon value seems more intuitive especially considering the data is now normalized to [0,1].
- For some reason concatenating the adversarial samples with the original data messed up the training. Each epoch now backpropogates on the original data and then the adversarial data. 
- No longer copying the model. The assumption is that the model will be trained from scratch anyway. 

I'm still testing this code on CelebA. It worked for CIFAR10, but I'm trying to find hyperparameters that work for Celeb,A which is a bit harder.

### CODE REVIEW

**Comment by asim29** [comment]
Tested the code on CelebA. It should be good to go, however, CelebA requires a much lower epsilon (0.01) compared to CIFAR10 which works fine even with an epsilon of 0.1. 

The minimum epsilon is bounded by the step_size which is currently 0.01. If someone needs to go lower they would have to change the step size to be lower as well.

**Review by sebszyller — APPROVED** [review]
